### PR TITLE
#6987: Theme switch mechanism/workflow for context pages

### PR DESCRIPTION
**Pull request by allyoucanmap**
## Description

This PR introduces support for custom theme inside context pages and a refactor on the Theme component.
It's possible to add a theme object or array configuration to the context configuration that will be converted to a style link inside the page after the main theme of mapstore to allow selector overrides and customizations.

The expected theme configuration of a context should be:

```json
"theme": {
    "id": "dark",
    "type": "link",
    "href": "dist/themes/dark.css"
};
```

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature
 - [x] Refactoring (no functional changes, no api changes)

## Issue

**What is the current behavior?**

#6987

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [x] No

## Other useful information
when we will allow the use of css variable in the context theme configuration we could extend the array with an additional entry

example

```js
theme: [
    {
        id: 'dark',
        type: 'link',
        href: 'dist/themes/dark.css'
    },
    {
        id: 'custom-variables',
        type: 'css-variables',
        variables: {
            primary: '#ff0000'
        }
    }
];
```

### CODE REVIEW

File: web/client/components/theme/ContextTheme.jsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+import React from 'react';
+import { createPortal } from 'react-dom';
+import PropTypes from 'prop-types';
+
+/**
+ * Theme component provider for the context pages
+ * @memberof components.theme
+ * @name ContextTheme
+ * @prop {array} theme a list of theme configurations to apply in a specific context
```
Comment (MV88) [inline web/client/components/theme/ContextTheme.jsx:18]:
```suggestion
 * @prop {object[]} theme a list of theme configurations to apply in a specific context
```

File: web/client/components/theme/ContextTheme.jsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+ * @example
+ * // add a css link tag in the context page
+ * <ContextTheme
+ *  theme={[{
+ *      id: 'dark',
+ *      type: 'link',
+ *      href: 'dist/themes/dark.css'
+ *  }]}/>
```
Comment (MV88) [inline web/client/components/theme/ContextTheme.jsx:27]:
```suggestion
 * <ContextTheme
 *  theme={[{
 *      id: 'dark',
 *      type: 'link',
 *      href: 'dist/themes/dark.css'
 *  },
 *  {
 *     id: 'custom-variables',
 *     type: 'css-variables',
 *     variables: {
 *         primary: '#ff0000'
 *    }
 *   }]}/>
```

File: web/client/components/theme/__tests__/ContextTheme-test.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+                    {
+                        id: 'dark',
+                        type: 'link',
+                        href: 'path/to/dark.css'
+                    }
+                ]}
+            />,
+            document.getElementById("container")
+        );
+        const contextStyles = document.querySelectorAll('[data-ms-context-theme]');
```
Comment (MV88) [inline web/client/components/theme/__tests__/ContextTheme-test.jsx:44]: can you test that a new link tag is added?

File: web/client/selectors/context.js [hunk 4]
```diff
@@ -95,3 +95,9 @@ export const contextMonitoredStateSelector = createSelector(
  * @param {object} state the app state
  */
 export const resourceSelector = ({context = {}} = {}) => context.resource;
+
+/**
+ * Get the theme configuration for the selected context
+ * @param {object} state the app state
+ */
+export const contextThemeSelector = ({context = {}} = {}) => castArray(context?.currentContext?.theme || []);
```
Comment (MV88) [inline web/client/selectors/context.js:103]: can you add a unit test for this?
maybe castArray is not necessary, but ok

**Review by MV88 — CHANGES_REQUESTED** [review]
please have a look at my comments,
just minor things

we need to improve ContextCreator documentation with the info for 
configuring properly the **theme** array property
should i do it in my PR [here](https://github.com/geosolutions-it/MapStore2/pulls)

File: web/client/components/theme/ContextTheme.jsx (hunk 2 above)
Comment (allyoucanmap) [reply web/client/components/theme/ContextTheme.jsx:27]: variable are not currently supported. I think we should improve the doc only when we add them

File: web/client/selectors/context.js (hunk 4 above)
Comment (allyoucanmap) [reply web/client/selectors/context.js:103]: about castArray it depends on how the theme will be stored, for now it's only a single object

**Review by MV88 — APPROVED** [review]

**Pull request merged**